java/client/src/main/java/glide/api/models/commands/LCSOptions.java

```diff
@@ -0,0 +1,76 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models.commands;
+
+import static glide.utils.ArrayTransformUtils.concatenateArrays;
+
+import lombok.Getter;
+import org.apache.commons.lang3.ArrayUtils;
+
+/**
+ * Optional arguments for {@link glide.api.commands.StringBaseCommands#lcs(String, String,
+ * LCSOptions)} command. <code>LEN</code> and <code>IDX</code> options are mutually exclusive.
+ *
+ * @see <a href="https://redis.io/commands/lcs/">redis.io</a>
+ */
+@Getter
+public final class LCSOptions {
+    /** Redis API keyword used to get the length of the LCS between strings. */
+    public static final String LENGTH_REDIS_API = "LEN";
+
+    /** Redis API keyword used to query indexes of match positions from each string in their LCS. */
+    public static final String IDX_REDIS_API = "IDX";
+
+    /**
+     * Redis API keyword used to restrict the list of matches to the ones of a given minimal length.
+     */
+    public static final String MINMATCHLEN_REDIS_API = "MINMATCHLEN";
+
+    /** Redis API keyword used to include the match length in the result of LCS. */
+    public static final String WITH_MATCHLEN_REDIS_API = "WITHMATCHLEN";
+
+    /**
+     * <code>true</code> if the LEN option is present, <code>false</code> if the IDX option is present
+     */
+    private final boolean isLEN;
+
+    /** The minimal match length value, <code>-1</code> if the option is not included */
+    private long minMatchLen = -1;
+
+    /** Indicates if the WITHMATCHLEN option should be included */
+    private final boolean withMatchLen;
+
+    /** Optional arguments constructor with the Length/IDX option. */
+    public LCSOptions(boolean isLEN) {
```

Comment:
Maybe we need to use a factory methods or a builder. I can't analyze what is better now, becase I don't know how command works (redis docs are misleading)